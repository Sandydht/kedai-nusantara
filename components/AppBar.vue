<template>
  <div id="appBar">
    <button @click="click">
      <svg xmlns="http://www.w3.org/2000/svg" class="h-6 w-6" fill="none" viewBox="0 0 24 24" stroke="currentColor">
        <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M4 6h16M4 12h16M4 18h16" />
      </svg>
    </button>
    <h1>Kedai Nusantara</h1>
    <nav-bar />
  </div>
</template>

<script>
import NavBar from '~/components/NavBar.vue'

export default {
  name: 'AppBar',
  components: {
    NavBar
  },
  emits: ['handleClick'],
  methods: {
    click () {
      this.$emit('handleClick')
    }
  }
}
</script>
