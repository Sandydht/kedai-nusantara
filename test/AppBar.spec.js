import { shallowMount } from '@vue/test-utils'
import AppBar from '~/components/AppBar.vue'

describe('AppBar.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(AppBar)
  })

  test('should display app title', () => {
    expect(wrapper.find('h1').text()).toBe('Kedai Nusantara')
  })

  test('should emits an event when menu button clicked', async () => {
    await wrapper.find('button').trigger('click')
    expect(wrapper.emitted()).toHaveProperty('handleClick')
  })

  test('has expected html structure', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
