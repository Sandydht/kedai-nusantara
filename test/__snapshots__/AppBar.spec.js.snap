// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar.vue has expected html structure 1`] = `
"<div id=\\"appBar\\"><button><svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" class=\\"h-6 w-6\\">
      <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M4 6h16M4 12h16M4 18h16\\"></path>
    </svg></button>
  <h1>Kedai Nusantara</h1>
  <nav-bar-stub></nav-bar-stub>
</div>"
`;
